src/enum.cr

```diff
@@ -509,7 +509,24 @@ abstract struct Enum
   # If multiple members match the same normalized string, the first one is returned.
   def self.parse?(string : String) : self?
     {% begin %}
-      case string.gsub('-', '_').camelcase.downcase
+      # FIXME: There is no `StringLiteral#bytesize` or any other adequate means
+      # to figure out how much space we actually need. Maybe some regex could
+      # work. For now just play it safe.
+      # FIXME: We might want to establish some upper limit in case a member name
+      # is exorbitantly long.
+
+      # The following is an optimized normalization. It is equivalent to
+      # `string.gsub('-', '_').camelcase.downcase` but does not allocate.
+      {% max_size = @type.constants.map(&.size).sort.last %}
```

Comment:
Polish (non blocking): we can compute the maximum size in one pass:

```suggestion
      {% max_size = @type.constants.reduce(0) { |a, e| e.size > a ? e.size : a } %}
```

Reply:
> we can compute the maximum size in one pass

I'm sure it's technically faster, it doesn't read as clearly and I'd be surprised if it resulted in a meaningful performance difference. `map.sort.last` isn't ideal (and I agree with @straight-shoota that adding some of the `Enumerable` conveniences to macros would be _fantastic_) but it's more glanceable than the `reduce` call.